### PR TITLE
Support annotations on BlobInfos

### PR DESCRIPTION
**Pull request by williammartin**
According to the image spec [1], descriptors should be able to contain
annotations. Support them on the returned BlobInfo.

[1] https://github.com/opencontainers/image-spec/blob/fc936c78346d017dcdbc6dd220a8ff60675ce3b5/descriptor.md

Signed-off-by: Will Martin <wmartin@pivotal.io>

### CODE REVIEW

**Comment by tscolari** [comment]
To add a bit more of context, this is important for the [Grootfs](https://github.com/cloudfoundry/grootfs) project. We're pushing forward a feature to support buildpacks based on OCI images, and we're using annotations to differentiate normal layers from the droplets.

**Comment by williammartin** [comment]
@mtrmac @runcom Any thoughts on this?

File: image/oci_test.go [hunk 1]
```diff
@@ -112,12 +122,14 @@ func TestManifestOCI1ManifestMIMEType(t *testing.T) {
 
 func TestManifestOCI1ConfigInfo(t *testing.T) {
 	for _, m := range []genericManifest{
-		manifestOCI1FromFixture(t, unusedImageSource{}, "oci1.json"),
```
Comment (mtrmac) [inline image/oci_test.go]: Why is this removed?

**Comment by mtrmac** [comment]
> I can’t resist feeling that this will inevitably become a mess

(Of course, now that the annotations are specified by OCI, we don’t have _that much_ choice: either expose the raw annotations, or tell users to use the OCI-specific libraries directly. The former is probably still a better choice, especially in the read-only form implemented by this PR.)

File: image/oci_test.go (hunk 1 above)
Comment (williammartin) [reply image/oci_test.go]: Amateur hour mistake. Thanks!

**Comment by williammartin** [comment]
@mtrmac In general we agree, the annotations are bound to be abused. However, like you said, OCI has specified them and this PR is useful to us. I'm hopeful we'll be able to remove our own dependence on annotations at some point but we do depend on them now.

**Comment by mtrmac** [comment]
:+1:

Thanks!

**Comment by runcom** [comment]
LGTM

**Comment by runcom** [comment]
(needs rebase)

**Comment by spikymonkey** [comment]
Thanks - branch rebased

**Comment by mtrmac** [comment]
Rebased and merged manually, per the conversation in #337.
